packages/report-flaky-tests/lib/github-api.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import * as github from '@actions/github';
+import * as unzipper from 'unzipper';
+
+/**
+ * Internal dependencies
+ */
+import type { FlakyTestResult } from './types';
+
+class GitHubAPI {
+	octokit: ReturnType< typeof github.getOctokit >;
+
+	constructor( token: string ) {
+		this.octokit = github.getOctokit( token );
+	}
+
+	get headBranch() {
+		return github.context.payload.workflow_run.head_branch;
+	}
+
+	get runURL() {
+		return github.context.payload.workflow_run.html_url;
+	}
+
+	async downloadReportFromArtifact(
+		artifactNamePrefix: string
+	): Promise< FlakyTestResult[] | undefined > {
+		const {
+			data: { artifacts },
+		} = await this.octokit.rest.actions.listWorkflowRunArtifacts( {
+			...github.context.repo,
+			run_id: github.context.payload.workflow_run.id,
+		} );
+
+		const matchArtifact = artifacts.find( ( artifact ) =>
+			artifact.name.startsWith( artifactNamePrefix )
+		);
+
+		if ( ! matchArtifact ) {
+			return undefined;
+		}
+
+		const download = await this.octokit.rest.actions.downloadArtifact( {
+			...github.context.repo,
+			artifact_id: matchArtifact.id,
+			archive_format: 'zip',
+		} );
+
+		const { files } = await unzipper.Open.buffer(
+			Buffer.from( download.data as Buffer )
```

Comment:
What type does `download.data` have? Would it be better to use a guard instead of always casting to `Buffer`?

Reply:
`download.data` has the type of `unknown`. I think that's an issue of the official typings of `@actions/github` and it should always be `Buffer`.

Reply:
Cool 👍 thanks for checking